Test.Activities.Automation.ActivityLib/Models/Activity/Services/SyncActivityService.cs

```diff
@@ -0,0 +1,246 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.SharePoint;
+using Test.Activities.Automation.ActivityLib.Models.Activity.Classes;
+using Test.Activities.Automation.ActivityLib.Models.Helpers;
+using Test.Activities.Automation.ActivityLib.Utils.Constants;
+
+namespace Test.Activities.Automation.ActivityLib.Models.Activity.Services.SyncActivityService
+{
+    public class SyncActivityService
+    {
+        private ILogger _logger;
+
+        public SyncActivityService(ILogger logger)
+        {
+            _logger = logger;
+        }
+
+        public void SyncActivities(IEnumerable<ActivityInfo> activities)
+        {
+            try
+            {
+                _logger?.LogInformation("Synchronizing activities");
+
+                if (activities == null)
+                {
+                    throw new Exception("Activities are null");
+                }
+
+                var now = DateTime.Now;
+                var minDate = new DateTime(now.Year, now.AddMonths(-1).Month, 1);
+                var maxDate = now.Date;
+
+                var ensureService = new EnsureService(_logger);
+
+                ensureService.CheckActivityDate(ref activities, minDate, maxDate);
+
+                using (var site = new SPSite(Constants.Host))
+                using (var web = site.OpenWeb(Constants.Web))
```

Comment:
You may initialize `SPWeb` only once: in the TimerJob. Sources, Services may take it as a constructors' parameter. In WCF Service you have `SPContext.Current.Web` (note: you do not need to dispose `SPContext` properties)